File: sample/sample-android/src/main/java/com/example/sampleandroid/view/model/UserScreen.kt
package com.example.sampleandroid.view.model

import androidx.compose.foundation.ScrollableColumn
import androidx.compose.material.Text
import androidx.compose.runtime.Composable
import kim.jeonghyeon.androidlibrary.compose.Screen
import kim.jeonghyeon.androidlibrary.compose.asValue
import kim.jeonghyeon.androidlibrary.compose.widget.Button
import kim.jeonghyeon.sample.viewmodel.UserViewModel
import kim.jeonghyeon.util.log

@Composable
fun UserScreen(model: UserViewModel) {
    Screen(model) {
        ScrollableColumn {
            val userDetail = model.user.asValue() ?: return@ScrollableColumn
            Text("Id : ${userDetail.id}")
            Text("Name : ${userDetail.name}")
            Button("Log Out") { model.onClickLogOut() }
        }
    }
}

// TODO reactive way.
//@Composable
//fun UserScreen2(model: UserViewModel2) {
//    Screen(model) {
//        ScrollableColumn {
//            val userDetail = model.user.asValue() ?: return@ScrollableColumn
//            Text("Id : ${userDetail.id}")
//            Text("Name : ${userDetail.name}")
//            Button("Log Out", model.click)
//        }
//    }
//}

File: sample/sample-native/kotlinIOS/screen/model/UserScreen.swift
//
//  SigninScreen.swift
//  kotlinIOS
//
//  Created by hyun kim on 04/09/20.
//  Copyright © 2020 hyun kim. All rights reserved.
//

import SwiftUI
import sample_base

func UserScreen(_ model: UserViewModel) -> some View {
    Screen(model) {
        ScrollableColumn {
            if let userDetail = model.user.asValue(viewModel: model) {
                Text("Id : \(userDetail.id!)")
                Text("Name : \(userDetail.name)")
                Button("Log Out") { model.onClickLogOut() }
            }
        }
    }
}
